mesh/meshdb.go

```diff
@@ -24,19 +24,21 @@ type layerMutex struct {
 // DB represents a mesh database instance
 type DB struct {
 	log.Log
-	blockCache         blockCache
-	layers             database.Database
-	blocks             database.Database
-	transactions       database.Database
-	contextualValidity database.Database
-	general            database.Database
-	unappliedTxs       database.Database
-	unappliedTxsMutex  sync.Mutex
-	blockMutex         sync.RWMutex
-	orphanBlocks       map[types.LayerID]map[types.BlockID]struct{}
-	layerMutex         map[types.LayerID]*layerMutex
-	lhMutex            sync.Mutex
-	exit               chan struct{}
+	blockCache            blockCache
+	layers                database.Database
+	blocks                database.Database
+	transactions          database.Database
+	contextualValidity    database.Database
+	general               database.Database
+	unappliedTxs          database.Database
+	inputVector           database.Database
+	unappliedTxsMutex     sync.Mutex
+	blockMutex            sync.RWMutex
+	orphanBlocks          map[types.LayerID]map[types.BlockID]struct{}
+	layerMutex            map[types.LayerID]*layerMutex
+	lhMutex               sync.Mutex
+	InputVectorBackupFunc func(id types.LayerID) ([]types.BlockID, error)
```

Comment:
does this need to be exported?

Reply:
either exported or create setter or pass it in the constructor.. do you think one of these is better?

Reply:
whatever you think is best in term of the dependancies.